File: 2020/GTBIT/L001_FoundationScratch/l001_Basics/l001Basics.java
import java.util.Scanner;
public class l001Basics{
    public static Scanner scn = new Scanner(System.in);

    public static int multiply(int a,int b){
        int c = a * b;
        return c;
    }
    
    public static int add(int a,int b){
        int c = a + b;
        return c;
    }
    
    public static int subtract(int a,int b){
        int c = b - a;
        return c;
    }
    
    public static int divide(int a,int b){
        int c = b / a;
        return c;
    }
    
    public static void printmessage(){
       System.out.println("see you soon!");
    }

    public static void gradingSystem(int marks){
        if(marks > 90){
            System.out.println("excellent");
        }else if(marks > 80){
            System.out.println("good");
        }else if(marks > 70){
            System.out.println("fair");
        }else if(marks > 60){
            System.out.println("meets expectations");
        }else{
            System.out.println("below par");
        }
    }

    public static void printNTime(int n){
        for(int i = 1; i <= n; i++){
            System.out.println("Hello!");
        }
    }

    public static void printTableOfN(int n){
        for(int i = 1;i <= 10;i++){
            System.out.println(n + " X " +  i + " = " + n * i);
        }
    } 

    public static boolean isPrime(int n){
        for(int i = 2; i*i <= n; i++){
            if(n % i == 0) return false;
        }
        return true;
    }

    public static void primeNumbers(){
        int t = scn.nextInt();
        for(int i = 1; i <= t; i++){
            int n = scn.nextInt();
            
            if(isPrime()){
                System.out.println("prime");
            }else{
                System.out.println("not prime");
            }
        }
    }

    public static void primeTillN(int n,int m){
        for(int i = n; i <= m; i++){
            if(isPrime(i)) System.out.println(i);
        }
    }

    public static int gcd(int a,int b){
    
        int divisor = a;
        int dividend = b;
        
        while(dividend % divisor != 0){
            int rem = dividend % divisor;
            dividend = divisor;
            divisor = rem;
        }
        
        return divisor;
    }
    
    public static int lcm(int a,int b,int g){
        return (a * (b / g));
    }

    public static int digits(int n){
        int count = 0;
        while(n != 0){
            count++;
            n /=10;
        }
        
        return count;
    } 
    
    public static int rotate(int n,int r){
        int len = digits(n);
        // r %= len;
        // if(r < 0) r += len;

        r = (r % len + len) % len;

        int mul = 1;
        int div = 1;
        
        for(int i=1;i<=len;i++){
            if(i <= r) div *= 10;
            else mul *= 10;
        }
        
        int a = n % div;
        int b = n / div;
        
        return a * mul + b;
    }
    
    
    public static void main(String[] args){
    //     System.out.println("Hello");
    //     System.out.println("Pepcoder");
    // int a = scn.nextInt();
    // int b = scn.nextInt();
    
    // int c = multiply(a,b);
    // System.out.println(multiply(a,b));
    // System.out.println(multiply(a,b));
    // System.out.println(add(a,b));
    // System.out.println(subtract(a,b));
    // printmessage();
    
    int n = scn.nextInt();
    // printNTime(n);
    printTableOfN(n);
   }
}

File: 2020/GTBIT/L001_FoundationScratch/l001_Basics/l001Basics.cpp
#include <iostream>
using namespace std;

int multiply(int a, int b)
{
    int c = a * b;
    return c;
}

int add(int a, int b)
{
    int c = a + b;
    return c;
}

int subtract(int a, int b)
{
    int c = b - a;
    return c;
}

int divide(int a, int b)
{
    int c = b / a;
    return c;
}

void printmessage()
{
    cout << "see you soon!" << endl;
}

void printNTime(int n)
{
    for (int i = 1; i <= n; i++)
    {
        cout << ("Hello!") << endl;
    }
}

void printTableOfN(int n)
{
    for (int i = 1; i <= 10; i++)
    {
        // cout << n << " X " << i << " = " << n * i << endl;
        cout << to_string(n) + " X " + to_string(i) + " = " + to_string(n * i) << endl;
    }
}

void printTableTillM(int m)
{
    for (int i = 1; i <= m; i++)
    {
        printTableOfN(i);
        cout << endl;
    }
}

void oddEven(int n){
    if(n % 2 == 0){
        cout <<to_string(n) + " -> even" << endl;
    }else{
        cout << to_string(n) + " -> odd" << endl;
    }
}

void checkNforOddEven(){
    int totalInputs; cin >> totalInputs;

    for(int i = 1;i <= totalInputs; i++){
        int number;
        cin >> number;

        oddEven(number);
    }
}


int countDigits(int n){
    int count = 0;
    while(n != 0){
        n /= 10;
        count++;
    }
    
    return count;
}

void reverse(int n){
    while(n != 0){
        int lastDigit = n % 10;
        n /= 10;
        
        cout << lastDigit << endl;
    }
}

int reverseNumber(int n){
    int ans = 0;
    while(n != 0){
        int lastDigit = n % 10;
        n /= 10;
        
        ans = ans * 10 + lastDigit;
    }

    return ans;
}

//digits-of-a-number
int powerOf10(int n){
    int power = 1;
    n /=10;
    while(n != 0){
        n /= 10;
        power *= 10;
    }
    
    return power;
}

void printDigits(int n){
    int power = powerOf10(n);
    while(power != 0){
        int num = n / power;
        int ld = num % 10;
        cout << ld <<endl;
        
        power /= 10;
    }
}

int gcd(int a,int b){
    
    int divisor = a;
    int dividend = b;
    
    while(dividend % divisor != 0){
        int rem = dividend % divisor;
        dividend = divisor;
        divisor = rem;
    }
    
    return divisor;
}

int lcm(int a,int b,int g){
    return (a * (b / g));
}

int main()
{
    // int a, b;
    // cin >> a >> b;
    // cout << multiply(a,b) << endl;
    // cout << multiply(a,b) << endl;
    // cout << multiply(a,b) << endl;
    // cout << multiply(a,b) << endl;
    // cout << multiply(a,b) << endl;
    // cout << add(a,b) << endl;
    // printmessage();

    // int n;
    // cin >> n;
    // printNTime(n);
    // printTableTillM(n);
    // printTableOfN(27);
    
    checkNforOddEven();

    return 0;
}
